extensions/wikia/AdEngine/js/lookup/amazonMatch.js

```diff
@@ -1,143 +1,40 @@
-		var key;
-		for (key in sizeMapping) {
-			if (sizeMapping.hasOwnProperty(key) && sizeMapping[key].indexOf(slotName) !== -1) {
-				return true;
-			}
-		}
-
-		return false;
-	}
-
-	function trackState(providerName, slotName, params) {
-		log(['trackState', amazonResponse, providerName, slotName], 'debug', logGroup);
-		var category,
-			eventName = 'lookup_error',
-			prices;
-
-		if (!isSlotSupported(slotName)) {
-			log(['trackState', 'Not supported slot', slotName], 'debug', logGroup);
-			return;
-		}
-		if (amazonResponse) {
-			eventName = 'lookup_success';
-		}
-		category = name + '/' + eventName + '/' + providerName;
-		if (params.amznslots) {
-			prices = params.amznslots.join(';');
-		}
-		adTracker.track(category, slotName, 0, prices || 'nodata');
-	}
-
-	function trackLookupEnd() {
-		var data = {};
-		Object.keys(sizeMapping).forEach(function (amazonSize) {
-			var pricePoint = bestPricePointForSize[amazonSize];
-			if (pricePoint) {
-				data['a' + amazonSize] = 'p' + pricePoint;
+		config = {
+			oasis: {
+				LEFT_SKYSCRAPER_2: ['1x6', '3x6'],
+				LEFT_SKYSCRAPER_3: ['1x6', '3x6'],
+				TOP_RIGHT_BOXAD: ['3x2', '3x6'],
+				HOME_TOP_RIGHT_BOXAD: ['3x2', '3x6'],
+				HUB_TOP_RIGHT_BOXAD: ['3x2', '3x6'],
+				TOP_LEADERBOARD: ['7x9', '9x2'],
+				HOME_TOP_LEADERBOARD: ['7x9', '9x2'],
+				HUB_TOP_LEADERBOARD: ['7x9', '9x2']
+			},
+			mercury: {
+				MOBILE_IN_CONTENT: ['3x2'],
+				MOBILE_PREFOOTER: ['3x2'],
+				MOBILE_TOP_LEADERBOARD: ['3x5']
 			}
-		});
-		adTracker.track(name + '/lookup_end', data || 'nodata', 0);
-	}
-
-	function onAmazonResponse() {
-		var allPricePointsForSize = {},
-			i,
-			len,
-			param,
-			m,
-			amazonSize,
-			amazonTier,
-			tokens;
-
-		amazonTiming.measureDiff({}, 'end').track();
-		tokens = win.amznads.getTokens();
-		log(['onAmazonResponse', tokens], 'debug', logGroup);
-		amazonResponse = true;
-
-		// First identify all correct amazon price points and record them in allPricePointsForSize
-		for (i = 0, len = tokens.length; i < len; i += 1) {
-			param = tokens[i];
-			m = param.match(amazonParamPattern);
-			if (m) {
-				amazonSize = m[1];
-				amazonTier = parseInt(m[2], 10);
-				if (!allPricePointsForSize[amazonSize]) {
-					allPricePointsForSize[amazonSize] = [];
-				}
-				allPricePointsForSize[amazonSize].push(amazonTier);
-			}
-		}
-
-		// Now select the minimal price point for each size we are interested in
-		Object.keys(bestPricePointForSize).forEach(function (amazonSize) {
-			var pricePoints = allPricePointsForSize[amazonSize];
-			if (pricePoints) {
-				bestPricePointForSize[amazonSize] = Math.min.apply(Math, pricePoints);
-			}
-		});
-
-		log(['onAmazonResponse - end', bestPricePointForSize], 'debug', logGroup);
-		trackLookupEnd();
-	}
-
-	function call() {
-		log('call', 'debug', logGroup);
+		},
+		rendered = false,
+		paramPattern = /^a([0-9]x[0-9])p([0-9]+)$/,
+		priceMap = {},
+		slots = [];
 
+	function call(skin, onResponse) {
```

Comment:
What happens when `onResponse` is `undefined`? [In Oasis](https://github.com/Wikia/app/pull/9345/files?diff=split#diff-2db853c9000d0443f3f8e46a9be8d024R28) you're passing only one parameter.